src/sql/platform/queryHistory/common/queryHistoryServiceImpl.ts

```diff
@@ -52,11 +52,14 @@ export class QueryHistoryService extends Disposable implements IQueryHistoryServ
 				const uri: URI = URI.parse(e.uri);
 				// VS Range is 1 based so offset values by 1. The endLine we get back from SqlToolsService is incremented
 				// by 1 from the original input range sent in as well so take that into account and don't modify
-				const text: string = _modelService.getModel(uri).getValueInRange(new Range(
-					e.queryInfo.selection[0].startLine + 1,
-					e.queryInfo.selection[0].startColumn + 1,
-					e.queryInfo.selection[0].endLine,
-					e.queryInfo.selection[0].endColumn + 1));
+				const text: string = e.queryInfo.selection.length > 0 ?
```

Comment:
just to double confirm e.queryInfo.selection can not be undefined or should be check that as well?

Reply:
It should be checked just to be safe, unless its shown that it is always available physically.

Reply:
Correct, the selection is non-optional (which is enforced by the strict null checking Anthony has been adding).

Reply:
Double checked this and the strict null checking isn't enabled for the queryModelService yet (which is where the events come from) so I'll add the check just to be safe.